### DISCUSSION TITLE
Using Array Data

### DISCUSSION THREAD

**Discussion by thoppa** [Budibase/budibase#10240] [Help] [answered]
I have a table with two columns: Start_Date and End_Date.

Ultimately, what I'm trying to achieve is to determine how many actual days in any given month are covered by the dates in the table.

For example: if Start_Date is 1/1/2023 and End_Date is 2/1/2023, then the January 2023 result would be 31 and February result 1.

I was able to build different arrays with these dates, and right now, both are stored as added columns.

````
const start = new Date($("Start_Date"));
const end = new Date($("End_Date"));
let nextFirst=new Date(start.getTime()), month, days={};
do { 
  year=nextFirst.getFullYear();
	month=nextFirst.getMonth();
  nextFirst.setMonth(month+1);nextFirst.setDate(1);
  days[`${year}-${String(1+month).padStart(2,"0")}`]=Math.round(((nextFirst<end?nextFirst:end)-start)/86400000);
	//days[month]=Math.round(((nextFirst<end?nextFirst:end)-start)/86400000);
  start.setTime(nextFirst.getTime());
} while(nextFirst<end)
	
return days;
````
This builds a really nice array with a count per month.  For example: {'2022-10':3}, for an entry with 3 days in October 2022.  If there are multiple months, it returns something that looks like this: {'2022-09':1,'2022-10':2}

The other column stores the actual dates that fall in between, returning `['2022-10-06','2022-10-07','2022-10-08']` in the first example above.

What I can't figure out how to do is use this data.  

I would love if I could store this information in a new table by year (with each row being a year and each column a month).  Meaning, the 2023  row would start with a January column, which would contain a number corresponding to the sum of all the January 2023 dates, and so on through the year. (I am assuming this means separate JS for each month/year combo, with an automation running when a row is added).  However, I would be nearly as happy with being able to present the data in the app with numbers and a bar graph for each month/year.

Any help or tips appreciated.

**Comment by melohagan** [comment]
Hey @thoppa 

You are using Formulas here?

If you want to have a table where you have a row for each year, and a column for each month, then you probably would just need to duplicate the formula for each of the twelve months, and filter for the particular month column, and the year value.

For something custom like this, it may even be more ideal to build a custom component? https://docs.budibase.com/docs/custom-plugin

There's a list of components here, one of which is a calendar component: https://github.com/Budibase/plugins

I'm not quite sure what you're hoping to achieve!

**Comment by thoppa** [comment]
I am effectively looking to have chart or table which details number of days per month by year.

So, like this:

            Jan.    Feb.  etc.

2023     4.        3       12
2024.     3.       2.       10

I also intend to average each column.

I can extract all this data using formulas, but referencing formula columns in another table doesn’t seem possible.  The gui design doesn’t really lend itself to creating tables (which makes sense, since you can just display db tables)

**Reply by melohagan** [reply]
Hey @thoppa 

You can reference formula columns in other tables by using a combination of static and dynamic formulas.
https://github.com/Budibase/budibase/discussions/5191

**Comment by thoppa** [comment, marked as answer]
None of these really answered my question, so I went in an entirely different direction.  

Rather than store this data in a db table (where I'd prefer), I used the markdown viewer element in the design and built a table.  There is a js function for each cell of the table that collects the dates and returns the sum. Not particularly elegant, but it gets the job done.
